lib/chess/core/move/rollback_services.rb

```diff
@@ -4,50 +4,20 @@ module Chess
   module Core
     module Move
       module RollbackServices
-        def roll_back
-          roll_back_piece_moved
-          @king_position_string = @pieces[@friends_team].king.position.algebraic.to_s
-
-          return ROLLBACK_SUCCES unless @piece_captured
-
-          find_cell(@piece_captured.position.algebraic).occup_by(@piece_captured)
-          add_captured_to_pieces
-          @piece_captured = nil
+        def roll_back(target_cell, piece_captured)
+          piece_moved = target_cell.occupant
+          piece_moved.roll_back_cell
+          piece_captured&.back_to_board
```

Comment:
in which scenario would `piece_captured` be nil? Is the safe guard operator (`&.`) really needed?

Reply:
is there a more explicit way to express this scenario you just said inside the method? Something like this:
```ruby
def roll_back(target_cell, piece_captured)
  piece_moved = target_cell.occupant
  piece_moved.roll_back_cell
  piece_captured.back_to_board if target_cell.cocupant.enemy?

  ROLLBACK_SUCCESS
end
```

Reply:
done, but let's keep ´piece_captured&.back_to_board´